x/tokenomics/keeper/keeper.go

```diff
@@ -1,46 +1,84 @@
 package keeper
 
 import (
+	"context"
 	"fmt"
 
 	"github.com/cometbft/cometbft/libs/log"
 	"github.com/cosmos/cosmos-sdk/codec"
 	storetypes "github.com/cosmos/cosmos-sdk/store/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	paramtypes "github.com/cosmos/cosmos-sdk/x/params/types"
-
+	suppliertypes "github.com/pokt-network/poktroll/x/supplier/types"
 	"github.com/pokt-network/poktroll/x/tokenomics/types"
 )
 
-type (
-	Keeper struct {
-		cdc        codec.BinaryCodec
-		storeKey   storetypes.StoreKey
-		memKey     storetypes.StoreKey
-		paramstore paramtypes.Subspace
-	}
-)
+// TokenomicsKeeperI is the interface contract that x/tokenomics's keeper implements.
+type TokenomicsKeeperI interface {
+	// GetAuthority returns the x/tokenomics module's authority.
+	GetAuthority() string
+
+	// SettleSessionAccounting is responsible for all of the post-session accounting
+	// necessary to burn, mint or transfer tokens depending on the amount of work
+	// done. The amount of "work done" complete is dictated by `sum` of `root`.
+	//
+	// ASSUMPTION: It is assumed the caller of this function validated the claim
+	// against a proof BEFORE calling this function.
+	// TODO_BLOCKER(@Olshansk): Is there a way to limit who can call this function?
+	SettleSessionAccounting(goCtx context.Context, claim suppliertypes.Claim)
+}
+
+// TokenomicsKeeper is the structure that implements the `TokenomicsKeeperI` interface.
+//
+// TODO_TECHDEBT(#240): See `x/auth/keeper.keeper.go` in the Cosmos SDK on how
+// we should refactor all our keepers. This keeper has started following a small
+// subset of those patterns.
+type TokenomicsKeeper struct {
```

Comment:
Generally, I think we should avoid grouping an interface and its implementation(s) in the same package to avoid potential dependency cycles.

Is this comment about #240 related to the name change (and/or packaging)? If not, I would push back against the name `TokenomicsKeeperI`, which would also be more difficult to justify if it were in a different package from `TokenomicsKeeper`.

Reply:
I was following the Cosmos v0.5 pattern. Example: https://github.com/cosmos/cosmos-sdk/blob/main/x/auth/keeper/keeper.go

I have no strong opinion. Since you'll be driving the migration next iteration, just let me know how you'd like me to structure & name this right now.

Reply:
@bryanchriswhite I started following the `0.5` pattern here: https://github.com/cosmos/cosmos-sdk/blob/main/x/auth/keeper/keeper.go

I anticipate this may or may not be modified during the migration work you and @h5law will be doing next week.

Lmk how you'd like for me to proceed in this specific instance for now.

Reply:
@bryanchriswhite In order to get this over the line, I've decided to remove the interface and follow the same naming convention we're using elsewhere for now.

Part of the cosmos v0.50 upgrade will be determining best practices moving forward.

Let me know if this approach works.